cmd/handler-utils.go

```diff
@@ -116,11 +116,10 @@ func extractMetadataFromHeader(header http.Header) map[string]string {
 	}
 	// Go through all other headers for any additional headers that needs to be saved.
 	for key := range header {
-		cKey := http.CanonicalHeaderKey(key)
-		if strings.HasPrefix(cKey, "X-Amz-Meta-") {
-			metadata[cKey] = header.Get(key)
+		if strings.HasPrefix(key, "X-Amz-Meta-") {
+			metadata[key] = header.Get(key)
```

Comment:
https://play.golang.org/p/hCJ6n1zLZQ 

```go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	headers := http.Header{}
	headers.Set("content-type", "test")
	headers["x-amz-meta-x-amz-key"] = []string{"vvv"}
	for k, v := range headers {
		fmt.Println(k, v)
	}
}
```

How do you handle a situation like this? is it expected that `headers` is not modified by the caller?